marketplaceItems/avatarCustomization101/appResources/appData/resources/js/avatarCustomization101_ui.js

```diff
@@ -0,0 +1,1326 @@
+        EVENT_UPDATE_AVATAR = APP_NAME + CONFIG.EVENT_UPDATE_AVATAR,
+        EVENT_CHANGE_AVATAR_TO_AVI_AND_SAVE_AVATAR = CONFIG.EVENT_CHANGE_AVATAR_TO_AVI_AND_SAVE_AVATAR,
+        EVENT_CHANGE_AVATAR_TO_AVI_WITHOUT_SAVING_AVATAR = CONFIG.EVENT_CHANGE_AVATAR_TO_AVI_WITHOUT_SAVING_AVATAR,
+        EVENT_RESTORE_SAVED_AVATAR = CONFIG.EVENT_RESTORE_SAVED_AVATAR,
+
+        // Material tab events
+        EVENT_UPDATE_MATERIAL = APP_NAME + CONFIG.EVENT_UPDATE_MATERIAL,
+        EVENT_UPDATE_BLENDSHAPE = APP_NAME + CONFIG.EVENT_UPDATE_BLENDSHAPE,
+        EVENT_UPDATE_FLOW = APP_NAME + CONFIG.EVENT_UPDATE_FLOW,
+
+        STRING_UPDATE_PROPERTY = CONFIG.MATERIAL_EVENTS_SUBTYPE.STRING_UPDATE_PROPERTY;
+
+    // Static TAB DATA
+    var STATIC_DATA = CONFIG.STATIC_DATA;
+    var TAB_LIST = STATIC_DATA.TAB_LIST;
+    var TAB_DATA = STATIC_DATA.TAB_DATA;
+
+    // Debug
+    var DEBUG = true;
+
+    // Components
+
+    // #region Tabs and Tab Layout
+
+    Vue.component('page-content', {
+        props: ['dynamicdata', 'staticdata'],
+        template: /* html */ `
+            <div>
+
+                <navigation 
+                    :dynamicdata="dynamicdata"
+                    :tablist="staticdata.TAB_LIST"
+                ></navigation>
+                
+                <tab-content-container
+                    :staticdata="staticdata"
+                    :dynamicdata="dynamicdata"
+                ></tab-content-container>
+
+            </div>
+        `
+    })
+
+    Vue.component('navigation', {
+        props: ['tablist', 'dynamicdata'],
+        template: /* html */ `
+            <nav>
+                <div class="nav nav-tabs nav-justified" id="nav-tab" role="tablist">
+
+                    <template v-for="tabName in tablist">
+                        
+                        <tab 
+                            :tabname="tabName"
+                            :activetabname="dynamicdata.state.activeTabName" 
+                            :isavienabled="dynamicdata.state.isAviEnabled"
+                        ></tab>
+
+                    </template>
+
+                </div>
+            </nav>
+        `
+    })
+
+    Vue.component('tab', {
+        props: ['tabname', 'activetabname', 'isavienabled'],
+        methods: {
+            switchTab(tabName) {
+
+                if (DEBUG) {
+                    console.log("Tab has been clicked: " + tabName);
+                }
+
+                EventBridge.emitWebEvent(JSON.stringify({
+                    type: EVENT_CHANGE_TAB,
+                    value: tabName
+                }));
+            }
+        },
+        computed: {
+            tabInfo() {
+
+                var tabName = this.tabname;
+                var isAviEnabled = this.isavienabled;
+
+                // set tab size depending on the size of the word
+                var flexGrowSize = tabName.length < 5 ? 1 : 2;
+                flexGrowSize = tabName.length < 10 ? flexGrowSize : 3;
```

Comment:
Yes, though I would still have this built in depending on the word letter length. I wanted to avoid doing a calculation by having the CSS do it for me :P